src/java.base/share/classes/java/lang/System.java

```diff
@@ -2660,6 +2662,20 @@ public StackWalker newStackWalkerInstance(Set<StackWalker.Option> options,
                                                       Continuation continuation) {
                 return StackWalker.newInstance(options, null, contScope, continuation);
             }
+
+            @Override
+            public int mismatchUTF8(String str, byte[] b, int fromIndex, int toIndex) {
+                byte[] encoded = str.isLatin1() ? str.value() : str.getBytes(UTF_8.INSTANCE);
+                if (false) {
+                    // Arrays.mismatch without the range checks (~3% faster micro getEntryHit)
+                    int aLength = encoded.length;
+                    int bLength = toIndex - fromIndex;
+                    int length = Math.min(aLength, bLength);
+                    int i = ArraysSupport.mismatch(encoded, 0, b, fromIndex, length);
+                    return (i < 0 && aLength != bLength) ? length : i;
+                }
+                return Arrays.mismatch(encoded, 0, encoded.length, b, fromIndex, toIndex);
```

Comment:
Leaving the ArraySupport.mismatch code here for now if anyone wants to investigate the ~3% regression introduced by the range checks in Arrays.mismatch